### ISSUE TITLE
Let's Encrypt: dns_transip_global_key should be yes or no

### ISSUE THREAD

**Issue by rbeumer** [home-assistant/addons#3876]
### Describe the issue you are experiencing

The renewal of my certificate is suddenly failing. I'm seeing the following error in the logs:

```
[09:18:58] INFO: Selected DNS Provider: dns-transip
[09:18:58] INFO: Use propagation seconds: 60
[09:18:58] INFO: Increasing DNS propagation limit for TransIP to at least 240 seconds.
[09:18:58] INFO: Detecting existing certificate type for [redacted]
Saving debug log to /var/log/letsencrypt/letsencrypt.log
[09:19:03] INFO: Existing certificate using 'ecdsa' key type.
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Renewing an existing certificate for [redacted]
Encountered exception during recovery: ValueError: dns_transip_global_key should have either 'yes' or 'no' as value
An unexpected error occurred:
ValueError: dns_transip_global_key should have either 'yes' or 'no' as value
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /var/log/letsencrypt/letsencrypt.log or re-run Certbot with -v for more details.
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

`dns_transip_global_key`  has been added to my addon config but it keeps failing and giving the same error message:

```
domains:
  - "*.[redacted]"
email: [redacted]
keyfile: privkey.pem
certfile: fullchain.pem
challenge: dns
dns:
  provider: dns-transip
  dns_transip_global_key: 'yes' (with or without quotes)
  transip_username: [redacted]
  transip_api_key: |
    [redacted]
keytype: rsa
```

Changing the value to no also doesn't change the behavior

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Which add-on are you reporting an issue with?

Let's Encrypt

### What is the version of the add-on?

5.2.10

### Steps to reproduce the issue

1. start the addon and check the logs

### System Health information

 There are currently no repairs pending 

### Anything in the Supervisor logs that might be useful for us?

```txt
2024-12-30 09:27:12.339 ERROR (SyncWorker_1) [supervisor.docker.manager] Container addon_core_letsencrypt is not running
2024-12-30 09:27:31.086 WARNING (MainThread) [supervisor.addons.options] Unknown option 'dns_transip_global_key' for Let's Encrypt (core_letsencrypt)
2024-12-30 09:27:31.087 WARNING (MainThread) [supervisor.addons.options] Option 'keytype' does not exist in the schema for Let's Encrypt (core_letsencrypt)
```

### Anything in the add-on logs that might be useful for us?

```txt
See the issue description
```

**Comment by agners** [comment]
Hm, I see the parameter has been introduced with https://github.com/home-assistant/addons/pull/3835, and is now essentially mandatory for DNS challenge of type `dns-transip`. Ideally I guess the script would just assume a default value if not given to prevent breaking existing configs. This also got discussed here: 
https://github.com/home-assistant/addons/pull/3855#issuecomment-2538993598

FWIW, the correct name of the add-on config option is `transip_global_key` (without the `dns_` prefix). Then it should work for you.

**Comment by Ascathon** [comment]
Yep, that did work (without dns_). Thank you.

**Comment by rrooggiieerr** [comment]
Hi! Just wanted to share that my existing `dns-transip` config broke due to the missing `transip_global_key`. I think it should use a default setting if the option is missing.

**Comment by eakoning** [comment]
Would it be possible to somehow intercept these kinds of messages (like `An unexpected error occurred:
ValueError: dns_transip_global_key should have either 'yes' or 'no' as value`) from various logs inside HA and render them on a dashboard in an overview so you can see something is failing hard? I really had no clue my cert was expired and could not be renewed of this and then all of a sudden everything falls apart. No way to use the app or bookmarked URLs to the application. All not working. That's kind of annoying.

What I'm asking is a simple way to get notified in your dashboard that something needs to be looked at ASAP, without having to grind through the logs of all moving parts, integrations, add-ons and whatnot every other day.

**Comment by hapklaar** [comment]
Same issue here. Would be great if errors like these would be shown as a HASS notification.

**Comment by GoetzGoerisch** [comment]
@agners should this still be adapted as described in your comment: https://github.com/home-assistant/addons/issues/3876#issuecomment-2565577197

**Comment by agners** [comment]
> [@agners](https://github.com/agners) should this still be adapted as described in your comment: [#3876 (comment)](https://github.com/home-assistant/addons/issues/3876#issuecomment-2565577197)

Yeah not sure, I am guessing that most people already fixed it by adding the config explicitly at this point. On the other hand, it probably takes up to 3 months for all users to actually run the add-on to get a new certificate, so it might still be helpful for some users. In either case, the change wouldn't hurt anyone, so I definitely would merge it.

**Comment by rrooggiieerr** [comment]
Don't forget that running the add-on without the correct config will fail silently until the certificate is expired and the user finds out the hard way. Especially since Let's Encrypt stopped sending expiration emails

**Comment by edwardhadders** [comment]
Hello, when i add transip_global_key: 'yes' its changing to transip_global_key: "yes" and giving me a error, anyone know a solution for this?

**Comment by rrooggiieerr** [comment]
For me `transip_global_key: 'yes'` is changed to `transip_global_key: "yes"` on saving the configuration.

I'm seeing this message in my log:
```
./run: line 279: dns_transip_global_key=no: command not found
```

**Comment by Ascathon** [comment]
Ah my bad. Yes I got that error now too. Haven't checked yet after all these updates.
The setting name has been reverted to dns_transip_global_key, as was the state before this issue was created.

**Comment by DennisBorCom** [comment]
> For me `transip_global_key: 'yes'` is changed to `transip_global_key: "yes"` on saving the configuration.
> 
> I'm seeing this message in my log:
> ```
> ./run: line 279: dns_transip_global_key=no: command not found
> ```

I have exactly the same issue. Did you het it working again?

**Comment by GoetzGoerisch** [comment]
Thank you for bringing this up and sorry that this is happening.

With the fix introduced in #3943  this was supposed to be fixed.
I see why this is happening and hope to provide a PR soon.

**Comment by GoetzGoerisch** [comment]
https://github.com/home-assistant/addons/pull/3958

**Comment by GoetzGoerisch** [comment]
Should be fixed with v5.4.1.
Please confirm @DennisBorCom @rrooggiieerr, thanks

**Comment by DennisBorCom** [comment]
> Should be fixed with v5.4.1. Please confirm [@DennisBorCom](https://github.com/DennisBorCom) [@rrooggiieerr](https://github.com/rrooggiieerr), thanks

I can confirm it is working again. Thanks for the quick fix!

**Comment by rrooggiieerr** [comment]
Yes, works for me too
